library/ssl_tls.c

```diff
@@ -1004,8 +1004,7 @@ static int ssl_populate_transform(mbedtls_ssl_transform *transform,
     !defined(MBEDTLS_SSL_EXPORT_KEYS) && \
     !defined(MBEDTLS_SSL_DTLS_CONNECTION_ID) && \
     !defined(MBEDTLS_DEBUG_C)
-    ssl = NULL; /* make sure we don't use it except for those cases */
-    (void) ssl;
+    (void) ssl; /* ssl is unused except for those cases */
```

Comment:
Could we actually use the compiler to help us do this?

```
#define ssl ssl_argument_must_not_be_accessed
```

Reply:
It's not really important that the function doesn't use `ssl` for other purposes. KISS.